go/graph/search.go

```diff
@@ -52,3 +52,62 @@ func (g *Graph) DijkstraSearch(start Node) []Path {
 	}
 	return paths
 }
+
+// Context for an AllPathSearch
+type allPathSearch struct {
+	// Graph
```

Comment:
I believe the field itself, named `Graph` with a type `*Graph`, is self explanatory and `// Graph` is unneeded here :stuck_out_tongue: